src/Dialogs.vala

```diff
@@ -32,12 +37,13 @@ namespace Gala {
         public string accept_label { get; set; }
         public string deny_label { get; set; }
 
-        public signal void response (uint response);
-
         const string PANTHEON_PORTAL_NAME = "org.freedesktop.impl.portal.desktop.pantheon";
         const string FDO_PORTAL_PATH = "/org/freedesktop/portal/desktop";
         const string GALA_DIALOG_PATH = "/io/elementary/gala/dialog";
 
+        [CCode (delegate_target = false)]
+        protected class OnResponseCallback on_response;
```

Comment:
Wouldn't it be more elegant/idiomatic to have a `protected virtual uint on_response (uint response)`

Reply:
done!